corehq/apps/users/views/__init__.py

```diff
@@ -891,8 +894,11 @@ def post_user_role(request, domain):
         }, status=400)
 
     response_data = role.to_json()
-    user_count = get_role_user_count(domain, role.couch_id)
-    response_data['hasUsersAssigned'] = user_count > 0
+    if role.is_commcare_user_default:
+        response_data["preventRoleDelete"] = True
+    else:
+        user_count = get_role_user_count(domain, role.couch_id)
+        response_data['preventRoleDelete'] = user_count > 0
```

Comment:
You might pull this out into a helper fn since it's basically the same as the block used above

Reply:
oh right, there's that `show_es_issue` business, eh whatever you feel like